### PR TITLE
Restore broadcast_key between restarts

### LINKED ISSUE #215: Save the last broadcast key between runs
The wemo stage seems to not accept connections as frequently anymore once broadcast events have been subscribed to

### PR DESCRIPTION
**Pull request by bdraco**
TLDR: This fixes battery powered BLE devices that sleep for long intervals

Some accessories will sleep for a long time and only send broadcasted events which makes them have very long connection intervals to save battery. Since we need to connect to get a new broadcast key we now save the broadcast key between restarts to ensure we can decrypt the advertisements coming in even though we cannot make a connection to the device during startup. When we get a disconnected event later we will try again to connect and the device will be awake which will trigger a full sync

Additional details: https://github.com/home-assistant/core/pull/81211

fixes #215

### CODE REVIEW

File: aiohomekit/controller/ble/pairing.py [hunk 1]
```diff
@@ -636,9 +647,13 @@ async def _async_set_broadcast_encryption_key(self) -> None:
...
         long_term_pub_key_bytes = bytes.fromhex(long_term_pub_key_hex)
-        self._broadcast_decryption_key = BroadcastDecryptionKey(
-            self._derive(long_term_pub_key_bytes, b"Broadcast-Encryption-Key")
+        broadcast_key_bytes = self._derive(
+            long_term_pub_key_bytes, b"Broadcast-Encryption-Key"
         )
+        self._broadcast_decryption_key = BroadcastDecryptionKey(broadcast_key_bytes)
+        if self._accessories_state and self.broadcast_key != broadcast_key_bytes:
+            self._accessories_state.broadcast_key = broadcast_key_bytes
+            self._callback_and_save_config_changed(self.config_num)
```
Comment (bdraco) [inline aiohomekit/controller/ble/pairing.py:656]: todo: check if we need this

Comment (bdraco) [reply aiohomekit/controller/ble/pairing.py:649]: we do

File: aiohomekit/controller/ble/pairing.py [hunk 2]
```diff
@@ -85,7 +85,10 @@
 
 logger = logging.getLogger(__name__)
 
-DISCOVER_TIMEOUT = 30
+# The discover timeout is how longer we will wait for and advertisement to be
+# received. If we don't get it in this time we will try again later since
+# the scanner is always running anyways.
```
Comment (bdraco) [inline aiohomekit/controller/ble/pairing.py:90]: Not much point in waiting for something that may never come if the device is sleeping

**Review by Jc2k — APPROVED** [review]

File: aiohomekit/controller/ble/pairing.py [hunk 3]
```diff
@@ -85,7 +85,10 @@
 
 logger = logging.getLogger(__name__)
 
-DISCOVER_TIMEOUT = 30
+# The discover timeout is how longer we will wait for and advertisement to be
```
Comment (bdraco) [inline aiohomekit/controller/ble/pairing.py:88]:
```suggestion
# The discover timeout is how long we will wait for and advertisement to be
```

**Comment by bdraco** [comment]
Still need to retest with IP accessories. I don't expect it will matter but good to check

**Comment by bdraco** [comment]
All good 👍

**Comment by bdraco** [comment]
Will push a tag, retest, and update the HA PR shortly
